lib/services/chat_bot_service.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:collection';
+import 'package:summer2022/exceptions/invalid_command_exception.dart';
+import 'package:summer2022/models/ApplicationFunction.dart';
+import 'package:summer2022/services/bases/chat_bot.dart';
+import 'package:summer2022/utility/RouteGenerator.dart';
+
+class ChatBotService implements ChatBot {
+  Map<SiteAreas, List<String>> ChatFunctions = HashMap();
```

Comment:
This should probably be private or a constant.
Also, you can just inline populate this, rather than calling `_populateChatFunctions`.
See: https://dart.dev/guides/language/language-tour#maps